ui/src/app/machines/views/MachineDetails/MachineStorage/CacheSetsTable/CacheSetsTable.tsx

```diff
@@ -0,0 +1,97 @@
+import { MainTable } from "@canonical/react-components";
+import React from "react";
+
+import DoubleRow from "app/base/components/DoubleRow";
+import TableHeader from "app/base/components/TableHeader";
+import { useTableSort } from "app/base/hooks";
+import { formatBytes } from "app/utils";
+import type { NormalisedStorageDevice as StorageDevice } from "../types";
+
+const getSortValue = (
+  sortKey: keyof StorageDevice,
+  storageDevice: StorageDevice
+) => storageDevice[sortKey];
+
+type Props = { cacheSets: StorageDevice[] };
+
+const CacheSetsTable = ({ cacheSets }: Props): JSX.Element => {
+  const { currentSort, sortRows, updateSort } = useTableSort(getSortValue, {
+    key: "name",
+    direction: "descending",
+  });
+  // TODO: update useTableSort to TS with generics
+  // https://github.com/canonical-web-and-design/maas-ui/issues/1869
+  const sortedCacheSets = sortRows(cacheSets) as StorageDevice[];
+
+  return (
+    <>
```

Comment:
Looks like you might be able to remove this fragment if this is just wrapping `<MainTable>`.

Reply:
You're right! Removed